flatflow/nemo/collections/nlp/data/language_modeling/megatron/gpt_sft_dataset.py

```diff
@@ -151,32 +151,52 @@ def __getitem__(self, idx):
         }
 
     def __sizeof__(self, idx):
-        return self.indexed_dataset[idx]["token_count"]
+        if isinstance(idx, np.int64):  # type: ignore[arg-type]
+            idx = idx.item()
+
+        if self.samples_mapping is not None:
+            assert idx < len(self.samples_mapping)
+            idx, _, _ = self.samples_mapping[idx]
+            if isinstance(idx, np.uint32):  # type: ignore[arg-type]
+                idx = idx.item()
+
+        assert idx < len(self.indexed_dataset)
+        auto_gen_idx = idx < 0
+        if auto_gen_idx:
```

Comment:
I think this line can be simplified to ``if idx < 0:`` since ``auto_gen_idx`` is used only once.

Reply:
That's correct. I just changed it as you suggested.
Thanks for the detailed review.